drivers/spi/Kconfig.nrfx

```diff
@@ -111,9 +115,12 @@ config SPI_1_NRF_ORC
 
 endif # SPI_1_NRF_SPI || SPI_1_NRF_SPIM || SPI_1_NRF_SPIS
 
-endif # SPI_1 && !I2C_1
+endif # SPI_1 && !I2C_1 && !(SOC_SERIES_NRF91X && UART_1_NRF_UARTE)
 
-if SPI_2
+# In Nordic SoCs, SPI and TWI peripherals with the same instance number
+# share certain resources and therefore cannot be used at the same time
+# (in nRF91 Series this limitation concerns UART peripherals as well).
```

Comment:
nit: UARTE, not UART

Reply:
Actually, it was intentional. I wanted to refer to UART as a general type of peripheral, like I did above with SPI and TWI (without going into details with SPIM, SPIS and so on). I was a bit afraid that this may appear ambiguous but I couldn't find a more accurate phrasing - in case I used UARTE here, the SPI and TWI above might be understood as referring to only the legacy kind of Nordic peripherals. And on the other hand, UARTE is still a UART (Universal asynchronous receiver/transmitter) with EasyDMA. Even the Product Specification sometimes uses the UART abbreviation for it, e.g. `TASKS_STARTRX` is described as "Start UART receiver".